src/pydvl/parallel/map_reduce.py

```diff
@@ -125,7 +139,17 @@ def __call__(
              The result of the reduce function.
         """
         seed_seq = ensure_seed_sequence(seed)
-        with Parallel() as parallel:
+
+        if isinstance(self.parallel_backend, JoblibParallelBackend):
+            backend = "loky"
+        elif isinstance(self.parallel_backend, RayParallelBackend):
+            backend = "ray"
+        else:
+            raise ValueError(
+                f"Unexpected parallel backend {self.parallel_backend.__class__.__name__}"
+            )
```

Comment:
I'm sure you don't like this either. :) But this makes user code unable to add parallel backends and use mapreduce with them. Not likely, but still, we could add a property to ParallelBackend to avoid this ifelse. Or add a third object to map backends to backends, but that seems overkill. Especially since joblib only also supports dask...

Reply:
For now, I changed this to raise an error if the passed parallel backend is not an instance of `JoblibParallelBackend`.

Reply:
If I understand it correctly, option 1 is a no-go. As you know, we have non-map-reduce jobs. Plus we need to signal running processes and so on. As to option 2, I'm not sure I follow. I would either leave this as it is, or add an attribute  `ParallelBackend.joblib_backend`. Both hacks, but the latter is at least local to the backends themselves